### ISSUE TITLE
[Feature Request]: Dialog, disable block-layer or need to select place in DOM where this is placed

### ISSUE THREAD

**Issue by timogeib** [SAP/ui5-webcomponents#7816]
### Feature Request Description

because of the way our component is embedded in another ui5 application,
we cannot use the dialog, as the ui5-block-layer will block everything, because it is placed too high up in the DOM and thus is moved outside of our own shadow-root.
This will break the z-index. We would need a new option on the dialog to disable the block-layer, so we can create our own or to be able to manipulate WHERE the ui5-static-area is created in the DOM

![image](https://github.com/SAP/ui5-webcomponents/assets/137267915/9d51935c-002e-4de8-9276-ee70261af676)

### Proposed Solution

Be able to disable the in-build block-layer to be able to create our own.

OR

Be able to select WHERE the ui5-static-area will be created in the DOM.

OR

Make sure that the ui5-static-area is NOT created OUTSIDE of the current SHADOW-ROOT

### Priority

Medium

### Privacy Policy

- [X] I’m not disclosing any internal or sensitive information.

**Comment by tsanislavgatev** [comment]
Hello @timogeib,

Can you please provide more information about your use case, so we can understand what your need is and try to help? A sample will be great!

The idea of the static-area is to be on top of everything else on the page with the correct CSS styles, regardless of the page structure and stacking context.

More information can be found here: 
https://sap.github.io/ui5-webcomponents/playground/?path=/docs/docs-development-deep-dive-and-best-practices--docs

Best Regards,
Tsanislav

**Comment by timogeib** [comment]
Our component is used in another ui5 application and our own component uses a shadow DOM to prevent overlaps in styles.
Because the ui5 webcomponent Dialog creates a "ui5-block-layer" inside a "ui5-static-area" on ROOT level of the DOM, which is OUTSIDE of our own shadow-root, the Z-index does not work and the block-layer always lays over all other components we create, EVEN the dialog popup of the ui5 webcomponent.

**Comment by timogeib** [comment]
if the root level static area is supposed to work even with shadow-roots, then please convert this request to a bug, as you can see in the screenshot, it does not work in our case.

**Comment by ilhan007** [comment]
As discussed offline, there re several options
1. obviously, not migrating to Dialog web component 
2. migrating to the Dialog web component 
- [available solution][requires effort from your side]
Usage of "portals" to tell angular app to render Dialog's content in other part of the DOM.
- [long-term solution][no effort from your side]
get rid of the static area and make use of the new standard "popover" API https://github.com/SAP/ui5-webcomponents/issues/7897

**Comment by ilhan007** [comment]
Hello @timogeib,  I wanted to update you on the issue as there are a lot of developments on UI5 Web Components side.


Recently, we announced **UI5 Web Components v2** at [UI5con2024](https://youtu.be/MSdGLG2zLy8?t=3246).

And there is one major feature, that is also related to the current issue - we adopted the new Popover API in version 2  that resolves the current issue. Learn more in the Popver API [blog post](https://sap.github.io/ui5-webcomponents/nightly/blog/releases/popover-api-in-v2/) on the topic:

In a nutshell, we no longer need to render the popovers in a static area. 
Instead, in version 2 popovers are rendered into the Shadow DOM of the web component.

**For example**
- If we take the Select in version 1: The Select web component is split into two - the input part and the dropdown, rendered in the static area
- If we take the Select in version 2: Everything is in the shadow DOM of the Select web component at one place.

**For example,** 
the Dialog's block layer is not high in the DOM, but it's in the Shadow DOM of the Dialog:

<img width="467" alt="Screenshot 2024-06-19 at 12 30 20" src="https://github.com/SAP/ui5-webcomponents/assets/15702139/640ac3c9-847c-4d48-8fcc-0c92cce470ed">
